src/main/scala/dev/kamu/cli/commands/PullCommand.scala

```diff
@@ -15,20 +16,35 @@ class PullCommand(
   metadataRepository: MetadataRepository,
   sparkRunner: SparkRunner,
   datasetIDs: Seq[String],
-  all: Boolean
+  all: Boolean,
+  recursive: Boolean
 ) extends Command {
   private val logger = LogManager.getLogger(getClass.getName)
 
   def run(): Unit = {
-    val datasets =
-      if (all)
-        metadataRepository.getAllDatasets()
-      else
-        datasetIDs
-          .map(DatasetID)
-          .map(metadataRepository.getDataset)
-
-    val numUpdated = datasets
+    val dependsOn: DatasetID => List[DatasetID] = {
+      if (all || recursive) {
+        metadataRepository.getDataset(_).dependsOn.toList
+      } else {
+        metadataRepository
+          .getDataset(_)
+          .dependsOn
+          .toList
+          .filter(datasetIDs.map(DatasetID).contains)
+      }
+    }
+
+    val plan = {
+      val depGraph = new DependencyGraph[DatasetID](dependsOn)
+      depGraph.resolve(datasetIDs.map(DatasetID).toList)
```

Comment:
Let's put this extra logic into a method in `MetadataRepository` e.g.:
```
def getDatasetsInDependencyOrder(ids: Seq[DatasetID], recursive: Boolean): List[Dataset]
```

Reply:
Done.